### PR TITLE
SBERDOMA-273 fullscreen mode for chessboard

### PR DESCRIPTION
**Pull request by leonid-d**

### CODE REVIEW

File: apps/condo/domains/property/components/BasePropertyForm/index.tsx [hunk 1]
```diff
@@ -94,7 +94,12 @@ const BasePropertyForm: React.FC<IPropertyFormProps> = (props) => {
                                             ({ getFieldsValue, setFieldsValue }) => {
                                                 const { map } = getFieldsValue(['map'])
                                                 return (
-                                                    <PropertyPanels mode='edit' map={map} updateMap={ map =>  setFieldsValue({ map })} />
+                                                    <PropertyPanels
+                                                        mode='edit'
+                                                        map={map}
+                                                        handleSave={handleSave}
+                                                        updateMap={ map =>  setFieldsValue({ map })}
```
Comment (pahaz) [inline apps/condo/domains/property/components/BasePropertyForm/index.tsx:101]:
```suggestion
                                                        updateMap={map =>  setFieldsValue({ map })}
```

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx [hunk 2]
```diff
@@ -62,30 +101,90 @@ export const BuildingAxisY: React.FC<IBuildingAxisYProps> = ({ floors }) => {
...
+    maximizableElement,
+}) => {
+    const intl = useIntl()
+    const RequestFullscreenMessage = intl.formatMessage({ id: 'FullscreenRequest' })
+    const ExitFullscreenMessage = intl.formatMessage({ id: 'FullscreenExit' })
+
+    let isFullscreen
+    let setIsFullscreen
+
+    try {
```
Comment (pahaz) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx:138]: It’s looks strange to wrap hook in try catch. You should use it inside hook

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx [hunk 3]
```diff
@@ -109,3 +208,57 @@ export function useHorizontalScroll (): React.RefObject<HTMLElement>{
...
+        result = 'webkitFullscreenElement'
+    } else {
+        throw new Error('This browser not supports fullscreenElement')
+    }
+    return result
+}
+
+const requestFullscreen = async (elem): Promise<void> => {
+    if (elem.requestFullscreen) {
+        await elem.requestFullscreen()
```
Comment (pahaz) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx:230]: Are you sure that we need browser level full screen ? As for me we just need to change display view

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelView.tsx [hunk 4]
```diff
@@ -13,98 +13,126 @@ import {
...
+                                                    disabled
+                                                >{section.name}</UnitButton>
+                                            </div>
+                                        )
+                                    })
+                                }
+                            </ScrollContainer>
                             {
-                                Builder.visibleSections.length > 0 ? <BuildingAxisY floors={Builder.possibleChosenFloors} /> : null
+                                <BuildingChooseSections Builder={Builder} refresh={refresh} maximizableElement={maximizableElement}></BuildingChooseSections>
```
Comment (pahaz) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelView.tsx:131]:
```suggestion
                                <BuildingChooseSections Builder={Builder} refresh={refresh} maximizableElement={maximizableElement}/>
```

**Comment by pahaz** [comment]
As for me, the result of the work should looks like reusable Fillscreen component which accept accept function as prop for passing full screen functions inside children. 

https://reactjs.org/docs/render-props.html

API like here: https://github.com/andrelmlins/react-fullscreen#examples

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx [hunk 5]
```diff
@@ -1,10 +1,48 @@
...
+    margin: -24px -24px 0;
+    padding: 12px 24px;
+    ${({ edit }) => (!edit ? `
+        display: none;
+        width: 100%;
+    ` : '')}
+
+
+    div:fullscreen & {
+        border-bottom: 1px solid #D9D9D9;
```
Comment (Dimitreee) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx:24]: use colors from constants

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx [hunk 6]
```diff
@@ -1,10 +1,48 @@
...
+        background: rgba(255,255,255, 0.9);
```
Comment (Dimitreee) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx:25]: use colors from constants

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx [hunk 7]
```diff
@@ -62,30 +101,90 @@ export const BuildingAxisY: React.FC<IBuildingAxisYProps> = ({ floors }) => {
...
 }
 
-export const BuildingChooseSections: React.FC<IBuildingChooseSectionsProps> = ({ Builder, refresh }) => {
+
+const fullcreen = {
+    left: '0',
+    right: '0',
+    bottom: '0',
+    padding: '12px 61px 12px 45px',
+    background: 'rgba(255, 255, 255, 0.9)',
```
Comment (Dimitreee) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx:113]: use colors from constants

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx [hunk 8]
```diff
@@ -62,30 +101,90 @@ export const BuildingAxisY: React.FC<IBuildingAxisYProps> = ({ floors }) => {
...
+    ZIndex: '1',
+    Position: 'fixed',
+    borderTop: '1px solid #D9D9D9',
```
Comment (Dimitreee) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelCommon.tsx:116]: color from constants

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx [hunk 9]
```diff
@@ -61,29 +71,45 @@ export const BuildingPanelEdit: React.FC<IBuildingPanelEditProps> = ({ map, upda
...
-                {
+            <FullscreenHeader edit={true}>
+                <Row style={{ paddingBottom: '39px', marginRight: '36px' }}>
+                    {property && <Col flex={0} style={{ marginTop: '10px' }}><b>{property.address}</b></Col>}
+                    <Col style={{ marginLeft: 'auto' }}>
+                        <Button
+                            key='submit'
+                            onClick={handleSave}
+                            type='sberPrimary'
+                            // loading={isLoading}
```
Comment (Dimitreee) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx:82]: remove commented code

File: apps/condo/domains/property/components/panels/index.tsx [hunk 10]
```diff
@@ -4,28 +4,33 @@ import { ResidentPanelEdit, ResidentPanelView } from './ResidentPanel'
...
+export const PropertyPanels: React.FC<IPropertyPanels> = ({ mode, map, updateMap, handleSave }) => {
     const intl = useIntl()
     const BuildingTabTitle = intl.formatMessage({ id: 'pages.condo.property.form.BuildingTabTitle' })
     const ResidentsTabTitle = intl.formatMessage({ id: 'pages.condo.property.form.ResidentsTabTitle' })
+    const maximizableElement = React.useRef(null)
     return (
         <Tabs defaultActiveKey='1'>
             <TabPane tab={BuildingTabTitle} key='1'>
-                <FocusContainer style={{ margin: 'initial', marginTop: '40px' }}>
+                <FocusContainer mode={mode} ref={maximizableElement} style={{ margin: 'initial', marginTop: '40px', background: '#fff' }}>
```
Comment (Dimitreee) [inline apps/condo/domains/property/components/panels/index.tsx:29]: use colors from constants

File: apps/condo/domains/common/components/FocusContainer.ts [hunk 11]
```diff
@@ -1,9 +1,18 @@
 import styled from '@emotion/styled'
 import { colors } from '../constants/style'
 
-export const FocusContainer = styled.div`
+export const FocusContainer = styled.div<{
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/FocusContainer.ts:4]: Why did focus container know smthng about mode and color?  Why did u change padding, this changes can break markup at ticket/[index].tsx

File: apps/condo/domains/common/components/containers/GlobalStyle.tsx [hunk 12]
```diff
@@ -152,6 +152,32 @@ export default function GlobalStyle () {
                   cursor: pointer;
                 }
               }
+
+              div.scroll-container {
```
Comment (pahaz) [inline apps/condo/domains/common/components/containers/GlobalStyle.tsx:156]: I think it's not a part of the common components.
@Dimitreee should we put it close to the FullScreen component?

Comment (pahaz) [reply apps/condo/domains/common/components/containers/GlobalStyle.tsx:156]: Is it possible inject it in FullscreenWrapper?

Comment (Dimitreee) [reply apps/condo/domains/common/components/containers/GlobalStyle.tsx:156]: Use css in js, please. Place this components near FullScreen. see apps/condo/domains/common/components/LinkWithIcon.tsx

**Review by Dimitreee — APPROVED** [review]

**Comment by Dimitreee** [comment]
rebase, please

**Review by dkoviazin — APPROVED** [review]
Ok

**Pull request merged**